src/main/java/com/google/cloud/spanner/pgadapter/statements/CopyStatement.java

```diff
@@ -0,0 +1,216 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.cloud.spanner.pgadapter.statements;
+
+import static com.google.cloud.spanner.pgadapter.parsers.copy.Copy.parse;
+
+import com.google.cloud.spanner.pgadapter.parsers.copy.CopyTreeParser;
+import com.google.cloud.spanner.pgadapter.utils.MutationBuilder;
+import com.google.spanner.v1.TypeCode;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.csv.CSVFormat;
+
+public class CopyStatement extends IntermediateStatement {
+
+  private static final String COLUMN_NAME = "column_name";
+  private static final String SPANNER_TYPE = "spanner_type";
+  private static final String CSV = "CSV";
+
+  private String tableName;
+  private List<String> copyColumnNames;
+  private String formatType;
+  private String nullString;
+  private char delimiterChar;
+  private char quoteChar;
+  private char escapeChar;
+  private boolean hasHeader;
+  private int formatCode;
+
+  private Map<String, TypeCode> tableColumns;
+  private MutationBuilder mutationBuilder;
+
+  public CopyStatement(String sql, Connection connection) throws SQLException {
+    super();
+    this.sql = sql;
+    this.command = parseCommand(sql);
+    this.connection = connection;
+    this.statement = connection.createStatement();
+  }
+
+  /** @return Mapping of table column names to column type. */
+  public Map<String, TypeCode> getTableColumns() {
+    return this.tableColumns;
+  }
+
+  /** @return CSVFormat for parsing copy data based on COPY statement options specified. */
+  public CSVFormat getParserFormat() {
+    CSVFormat format = CSVFormat.POSTGRESQL_TEXT;
+    if (this.formatType != null && this.formatType.equalsIgnoreCase("CSV")) {
+      format = CSVFormat.POSTGRESQL_CSV;
+    }
+    if (this.nullString != null && !this.nullString.isEmpty()) {
```

Comment:
nit: You can use `com.google.common.base.Strings.isNullOrEmpty`

Reply:
Done.